esx_service/vmdk_ops.py

```diff
@@ -583,7 +583,7 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     path = get_vol_path(datastore, tenant_name)
     logging.debug("executeRequest %s %s", tenant_name, path)
     if path is None:
-        return err("Failed to initialize volume path {0}".format(path))
+        return err("%s is not writable and hence %s volume creation failed" % (datastore, full_vol_name.split('@')[0]))
```

Comment:
This change assumes path == Null means Datastore is not writable. `get_vol_path` could be returning empty path due to other errors. You may want to return error from `get_vol_path`